frontend/src/components/SearchBarFilter.tsx

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable array-callback-return */
+/* eslint-disable @typescript-eslint/strict-boolean-expressions */
+import React, { useState } from 'react'
+import { Tag } from './Tag'
+import './searchbar.css'
+import { type Friend } from './PlanForm'
+
+interface Props {
+  items: Friend[]
+  handleSelectChange: (friend: Friend) => void
+  handleRemoveTag?: (friend: Friend[]) => void
+  numbOfSelect?: number
+  setSelectError?: (friend: Friend) => void
+}
+
+export const SearchBarFilter: React.FC<Props> = ({
+  items,
+  handleSelectChange,
+  handleRemoveTag,
+  numbOfSelect,
+  setSelectError
+}) => {
+  const [query, setQuery] = useState<string>()
+  const [friend, setItems] = useState<Friend[]>([])
+  const [select, setSelected] = useState<Friend[]>([])
+
+  const handleQuery = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    const value = event.target.value
+    setQuery(value)
+
+    const filteredItems: Friend[] = items.filter(
+      obj => obj.firstName.toLowerCase().includes(value)
+    )
+    setItems(filteredItems)
+  }
+
+  const handleSelect = (friend: Friend): void => {
+    if (numbOfSelect &&
```

Comment:
generally im a little wary of using the js truthy shorthand, since expressions like `if (numbOfSelect) {}` may not evaluate the way you think it does for numbers and strings. `0` and `''` empty string both evaluate to false, which may be incorrect behavior if your intent is to check "is this variable defined"/"is this variable null or undefined"

Reply:
ok I just made `numOfSelect` a required prop to prevent this